docs/tutorials/contracts/guest-book.md

```diff
@@ -0,0 +1,993 @@
+---
+id: guest-book
+title: Build a Guest Book
+sidebar_label: Build a Guest Book
+---
+
+In this section we will create a starter app built with an [AssemblyScript] backend and a [React] frontend that allows users to sign in with [NEAR] and add a message to the guest book.
+
+
+## Intro notes 
+
+We will be referencing branches of a NEAR project called `guest-book`. 
+
+Each step will be it's own branch, so you can follow along or jump ahead to see the complete code.
+
+You need `near-cli` installed globally. Here's how:
+
+    npm install --global near-cli
+
+This will give you the `near` [CLI] tool. Ensure that it's installed with:
+
+    near --version
+
+Next, clone the repo [here]. We will start with the `boilerplate` branch, and slowly progress towards the master branch with the completed code. 
+
+Alternatively, create a basic NEAR React project with package [create-near-app] using the command:
+
+    yarn create near-app path/to/my-near-react-app
+
+Just adjust the folder structure to match the branch we will be referencing.
+
+
+### Local Setup
+
+To run this project locally:
+
+1. Prerequisites: Make sure you have [Node.js](https://nodejs.org) ≥ 12 installed, then use it to install [yarn]: `npm install --global yarn` (or just `npm i -g yarn`)
+2. Install dependencies: `yarn install` (or just `yarn`)
+3. Run the local development server: `yarn dev` (see `package.json` for a
+   full list of `scripts` you can run with `yarn`)
+
+Now you'll have a local development environment backed by the NEAR _TestNet_. Running `yarn dev` will tell you the URL you can visit in your browser to see the app.
+
+
+### Exploring The Code
+
+1. The backend code (smart contract) lives in the `/assembly` folder. This code deploys to
+   the NEAR blockchain when you run `yarn deploy:contract`  – [learn more
+   about NEAR smart contracts][smart contract docs].
+2. The frontend code lives in the `/src` folder. A great place to start exploring UI code 
+   is in `/src/index.js`, where you can see how the frontend connects to the NEAR blockchain.
+3. Tests: there are different kinds of tests for the frontend and backend. The
+   backend code gets tested with the [asp] command for running the backend
+   AssemblyScript tests, and [jest] for running frontend tests. You can run
+   both of these at once with `yarn test`.
+
+Both contract and client-side code will auto-reload as you update source files.
+
+
+
+## Contract
+
+If we imagine the blockchain to be a sort of database, then smart contracts function like an ORM or API with read/write access to the blockchain. 
+
+The methods that do _NOT_ update the state of your application are called _view_ methods. Those that do are called _call_ or _change_ methods.
+
+
+## Contract: Develop 
+
+For this section we will be working in the `assembly/` folder of your project, and all code is written in TypeScript.
```

Comment:
```suggestion
For this section we will be working in the `assembly/` folder of your project, and all code is written in AssemblyScript.
```

Its important that we make sure we distinguish TS from AS as they may seem similar they are very different.